src/psmove.c

```diff
@@ -303,6 +312,44 @@ static int psmove_remote_disabled = 0;
 static int psmove_num_open_handles = 0;
 
 
+static int
+psmove_fscanf_c(FILE *fp, const char *fmt, ...)
+{
+    va_list args;
+    va_start(args, fmt);
+#if defined(_WIN32)
+    _locale_t loc = _create_locale(LC_NUMERIC, "C");
+    int result = _vfscanf_l(fp, fmt, loc, args);
+    _free_locale(loc);
+#else
+    locale_t old = uselocale(newlocale(LC_NUMERIC_MASK, "C", (locale_t) 0));
+    int result = vfscanf(fp, fmt, args);
+    freelocale(uselocale(old));
+#endif
+    va_end(args);
+    return result;
+}
+
+static int
+psmove_fprintf_c(FILE *fp, const char *fmt, ...)
+{
+    va_list args;
+    va_start(args, fmt);
+
+#if defined(_WIN32)
+    _locale_t loc = _create_locale(LC_NUMERIC, "C");
+    _vfprintf_l(fp, fmt, loc, args);
+    _free_locale(loc);
+#else
+    locale_t old = uselocale(newlocale(LC_NUMERIC_MASK, "C", (locale_t) 0));
+    int result = vfprintf(fp, fmt, args);
+    freelocale(uselocale(old));
+#endif
+
+    va_end(args);
+    return result;
+}
+
```

Comment:
No error checking here. I don't love that, but it might not be that big of a deal.

Reply:
Error checking for `psmove_fscanf_c()` is done by checking the return value against `PSMOVE_MAGNETOMETER_CALIBRATION_FSTRING_FIELDS` on the call site.

Error checking for `psmove_fprintf_c()` isn't done at the moment (as it returns the number of bytes written, not how many fields have been written).

Reply:
(or did you mean error checking for `_create_locale`, `_free_locale`, `uselocale`, `newlocale` and `freelocale`?)

Reply:
Yes, that's what I meant.

Reply:
Added error checking for `newlocale()` now. The other functions seem to not be able to fail. Also, the Win32 APIs seem to only be available when linking against the Universal C Runtime (UCRT), so special-cased this, with a fallback (e.g. mingw-w64) to just the normal vfprintf/vfscanf functions. Eventually we should look into building against the UCRT on Windows. However, it could be that MSVCRT doesn't even have issues with locales?

Reply:
On  the off-chance that the whole thing is unnecessary with the MSVCRT, I will look into this later today.

Not sure about the UCRT issue, though. Doesn't that just complain about `_vfscanf_l` not being available? `_fscanf_l` should be fine. I mean, we could always just do all the platform-dependent special handling for that one particular format string we are dealing with and not try to generalize it.

Reply:
Interesting that `_fscanf_l` would be available, but that seems to be what the mingw headers tell me too.. weird.

Tried to take that into account in d37b68d523bf20b53b6da49e18dd9a381063357e (but untested, as I don't have a Windows build machine around, but CI should catch it).